### PR TITLE
Use FUS in AppCenterDistributeV3

### PR DESCRIPTION
**Pull request by annakocheshkova**
**Task name**: AppCenterDistributeV3

**Description**: Switch to using FUS for uploading the file.

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) N

**Checklist**:
- [ ] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [X] Checked that applied changes work as expected

Note: unit tests are in a separate PR. No need to bump task version yet.

### CODE REVIEW

File: Tasks/AppCenterDistributeV3/appcenterdistribute.ts [hunk 1]
```diff
@@ -91,10 +102,10 @@ function responseHandler(defer, err, res, body, handler: () => void) {
     handler();
 }
 
-function beginReleaseUpload(apiServer: string, apiVersion: string, appSlug: string, token: string, userAgent: string, buildVersion: string): Q.Promise<UploadInfo> {
+function beginReleaseUpload(apiServer: string, apiVersion: string, appSlug: string, token: string, userAgent: string, buildVersion: string): Q.Promise<any> {
```
Comment (MatkovIvan) [inline Tasks/AppCenterDistributeV3/appcenterdistribute.ts]: Can we avoid `any` here?

File: Tasks/AppCenterDistributeV3/appcenterdistribute.ts [hunk 2]
```diff
@@ -114,59 +125,116 @@ function beginReleaseUpload(apiServer: string, apiVersion: string, appSlug: stri
     request.post(requestOptions, (err, res, body) => {
         responseHandler(defer, err, res, body, () => {
             let response = JSON.parse(body);
-            let uploadInfo: UploadInfo = {
-                upload_id: response['upload_id'],
-                upload_url: response['upload_url']
+            if (!response.package_asset_id || (response.statusCode && response.statusCode !== 200)) {
```
Comment (MatkovIvan) [inline Tasks/AppCenterDistributeV3/appcenterdistribute.ts]: All 2xx are valid codes

File: Tasks/AppCenterDistributeV3/appcenterdistribute.ts [hunk 3]
```diff
@@ -263,6 +331,27 @@ function updateRelease(apiServer: string, apiVersion: string, appSlug: string, r
...
+function getReleaseId(apiServer: string, apiVersion: string, appSlug: string, releaseId: string, token: string, userAgent: string): Q.Promise<any> {
+    tl.debug("-- Getting release id.");
+    let defer = Q.defer<Release>();
+    let getReleaseUrl: string = `${apiServer}/${apiVersion}/apps/${appSlug}/uploads/releases/${releaseId}`;
+    tl.debug(`---- url: ${getReleaseUrl}`);
+
+    let headers = {
+        "X-API-Token": token,
+        "User-Agent": userAgent,
+        "internal-request-source": "VSTS"
```
Comment (MatkovIvan) [inline Tasks/AppCenterDistributeV3/appcenterdistribute.ts:346]: It's still VSTS? (not AzDO)

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

File: Tasks/AppCenterDistributeV3/appcenterdistribute.ts [hunk 4]
```diff
@@ -530,14 +622,23 @@ async function run() {
...
         // Begin release upload
-        let uploadInfo: UploadInfo = await beginReleaseUpload(effectiveApiServer, effectiveApiVersion, appSlug, apiToken, userAgent, buildVersion);
-
-        // Perform the upload
-        await uploadRelease(uploadInfo.upload_url, app, userAgent);
-
-        // Commit the upload
-        let releaseId = await commitRelease(effectiveApiServer, effectiveApiVersion, appSlug, uploadInfo.upload_id, apiToken, userAgent);
-
+        let uploadInfo: any = await beginReleaseUpload(effectiveApiServer, effectiveApiVersion, appSlug, apiToken, userAgent, buildVersion);
```
Comment (MatkovIvan) [inline Tasks/AppCenterDistributeV3/appcenterdistribute.ts]: I guess it's not "any"

**Review by MatkovIvan — APPROVED** [review]
Only nitpick. The code looks good in general, but I didn't test it personally

**Review by Jamminroot — APPROVED** [review]

**Pull request merged**